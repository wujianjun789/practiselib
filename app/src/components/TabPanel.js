/**
 * Created by a on 2017/10/13.
 */
import React,{Component} from 'react';

import '../../public/styles/tabPanel.less';

export default class TabPanel extends Component{
    constructor(props){
        super(props);
        const {data, activeId} = props;
        this.state = {
            activeId: activeId != undefined?activeId:(data && data.length?data[0].id:"")
        }
        this.tabClick = this.tabClick.bind(this);
    }

    tabClick(id, index){
        const {data} = this.props;
        this.setState({activeId:data[index].id}, ()=>{
            this.props.tabClick && this.props.tabClick(id, index);
        });
    }

    render(){
        const {className, data} = this.props;
        const {activeId} = this.state;

        return <div className={"tab-panel "+(className?className:"")}>
            <ul className="tab-header">
                {
                    data && data.map((item, index)=>{
                        return <li key={item.id} className={item.id==activeId?"active":""} onClick={()=>{this.tabClick(item.id, index)}}>{item.title}</li>
                    })
                }
            </ul>
            <div className="tab-content">
                {
                    this.props.children
                }
            </div>
        </div>
    }
}