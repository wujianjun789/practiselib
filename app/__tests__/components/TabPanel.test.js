/**
 * Created by a on 2017/10/16.
 */
import React from 'react';
import renderer from 'react-test-renderer'
import { shallow } from 'enzyme';
import TabPanel from '../../src/components/TabPanel';

const data = [
    {id:1, title:'shanghai'},
    {id:2, title:'beijing'},
    {id:3, title:'guangzhou'},
]
test('tabPanel renders', ()=>{

    const component = renderer.create(
        <TabPanel data={data} activeId={1}/>
    )

    let tabPanel = component.toJSON();
    expect(tabPanel).toMatchSnapshot();
})

test('tabPanel render', ()=>{
    const component = shallow(<TabPanel data={data} activeId={1}/>)
    expect(component.find('.tab-panel').length).toEqual(1);
    expect(component.find('.tab-panel').find('.tab-header').length).toEqual(1);
    expect(component.find('.tab-panel').find('.tab-header').find('li').length).toEqual(3);
    expect(component.find('.tab-panel').find('.tab-content').length).toEqual(1);
})

test('tabPanel header click', ()=>{
    const click = jest.fn();
    const component = shallow(<TabPanel data={data} activeId={1}/>)

    expect(component.find('.tab-panel').find('.tab-header').at(0).find('.active').length).toEqual(1);
    component.find('.tab-panel').find('.tab-header').at(1).simulate('click');
    console.log("activeId:",component.state('activeId'));
    // expect(component.find('.tab-panel').find('.tab-header').at(0).find('.active').length).toEqual(1);
})
