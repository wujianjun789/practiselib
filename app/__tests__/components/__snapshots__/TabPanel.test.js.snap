exports[`test tabPanel renders 1`] = `
<div
  className="tab-panel ">
  <ul
    className="tab-header">
    <li
      className="active"
      onClick={[Function]}>
      shanghai
    </li>
    <li
      className=""
      onClick={[Function]}>
      beijing
    </li>
    <li
      className=""
      onClick={[Function]}>
      guangzhou
    </li>
  </ul>
  <div
    className="tab-content" />
</div>
`;
